### ISSUE TITLE
[Sparse] Resolve macro conflicts between Pytorch and DGL loggers.

### ISSUE THREAD

**Issue by czkkkkkk** [dmlc/dgl#4943]
## 🚀 Feature
The sparse library has macro redefinition problems when including headers from Pytorch and DGL at the same time. Typical macros are `LOG`,  `CHECK` and `CHECK_EQ`. The current workaround is to include all DGL headers in one file and undef these macros. We need a more elegant way to resolve the problem.

**Comment by jermainewang** [comment]
As we are switching to `TORCH_CHECK` entirely, is this still an issue @czkkkkkk ?

**Comment by czkkkkkk** [comment]
> As we are switching to TORCH_CHECK entirely, is this still an issue @czkkkkkk ?

Yes. It is still an issue because it is caused by the macro conflicts between DGL headers and Pytorch headers.
